Parser/Device/AbstractDeviceParser.php

```diff
@@ -81,1300 +81,1304 @@ abstract class AbstractDeviceParser extends AbstractParser
+        'DL'  => 'Dell',
+        'DE'  => 'Denver',
+        'DS'  => 'Desay',
+        'DW'  => 'DeWalt',
+        'DX'  => 'DEXP',
+        '8D'  => 'DF',
+        'DG'  => 'Dialog',
+        'DI'  => 'Dicam',
+        'D4'  => 'Digi',
+        'D3'  => 'Digicel',
+        'DH'  => 'Digihome',
+        'DD'  => 'Digiland',
+        'Q0'  => 'DIGIFORS',
+        'DQ'  => 'DISH',
+        '9D'  => 'Ditecma',
+        'D2'  => 'Digma',
+        '1D'  => 'Diva',
+        'D6'  => 'Divisat',
+        'X6'  => 'DIXON',
+        '5D'  => 'DING DING',
+        'DM'  => 'DMM',
+        'DN'  => 'DNS',
+        'DC'  => 'DoCoMo',
+        'DF'  => 'Doffler',
+        'D9'  => 'Dolamee',
+        'DO'  => 'Doogee',
+        'D0'  => 'Doopro',
+        'DV'  => 'Doov',
+        'DP'  => 'Dopod',
+        'JQ'  => 'Doppio',
+        'DR'  => 'Doro',
+        'ZD'  => 'DORLAND',
+        'D8'  => 'Droxio',
+        'DJ'  => 'Dragon Touch',
+        'DY'  => 'Dreamgate',
+        'DU'  => 'Dune HD',
+        'UD'  => 'DUNNS Mobile',
+        'EB'  => 'E-Boda',
+        'EJ'  => 'Engel',
+        '2E'  => 'E-Ceros',
+        'E8'  => 'E-tel',
+        'EP'  => 'Easypix',
+        'EQ'  => 'Eagle',
+        'EA'  => 'EBEST',
+        'YC'  => 'EBEN',
+        'E4'  => 'Echo Mobiles',
+        'ES'  => 'ECS',
+        '35'  => 'ECON',
+        'ZZ'  => 'ecom',
+        'E6'  => 'EE',
+        'GW'  => 'EGL',
+        'EK'  => 'EKO',
+        'EY'  => 'Einstein',
+        'EM'  => 'Eks Mobility',
+        'UE'  => 'Ematic',
+        '4K'  => 'EKT',
+        '7E'  => 'ELARI',
+        '03'  => 'Electroneum',
+        'Z8'  => 'ELECTRONIA',
+        'L0'  => 'Element',
+        'EG'  => 'Elenberg',
+        'EL'  => 'Elephone',
+        'JE'  => 'Elekta',
+        '4E'  => 'Eltex',
+        'ED'  => 'Energizer',
+        'E1'  => 'Energy Sistem',
+        '3E'  => 'Enot',
+        '8E'  => 'Epik One',
+        'XP'  => 'Epson',
+        'E7'  => 'Ergo',
+        'EC'  => 'Ericsson',
+        '05'  => 'Erisson',
+        'ER'  => 'Ericy',
+        'EE'  => 'Essential',
+        'E2'  => 'Essentielb',
+        '6E'  => 'eSTAR',
+        'EN'  => 'Eton',
+        'ET'  => 'eTouch',
+        '1E'  => 'Etuline',
+        'EU'  => 'Eurostar',
+        '4J'  => 'Eurocase',
+        'E9'  => 'Evercoss',
+        'EV'  => 'Evertek',
+        'E3'  => 'Evolio',
+        'EO'  => 'Evolveo',
+        '0Q'  => 'Evoo',
+        '5U'  => 'EVPAD',
+        'E0'  => 'EvroMedia',
+        'XE'  => 'ExMobile',
+        '4Z'  => 'Exmart',
+        'EH'  => 'EXO',
+        'EX'  => 'Explay',
+        'E5'  => 'Extrem',
+        'EF'  => 'EXCEED',
+        'QE'  => 'EWIS',
+        'EI'  => 'Ezio',
+        'EZ'  => 'Ezze',
+        'UF'  => 'EYU',
+        '5F'  => 'F150',
+        'F6'  => 'Facebook',
+        'FA'  => 'Fairphone',
+        'FM'  => 'Famoco',
+        '17'  => 'FarEasTone',
+        '9R'  => 'FaRao Pro',
+        'FB'  => 'Fantec',
+        'FAC' => 'Facetel',
```

Comment:
Please note that the 2-character shortcodes have ended.

Can we transfer $deviceBrands to YML file?

Reply:
I actually don't see much benefit in moving this to a yml file. It will actually only add ann additional yml parsing, which might slow down the detection more.

Regarding three character shortcodes:
I need to check if Matomo is able to handle them properly before we can merge and release a new version. At least the database is able to store them already, but not sure if the code expects two chars somewhere. Need to investigate that. Meanwhile it might be better to focus on adding devices for already existing brands, as we can merge such PRs.

Reply:
OK, I will move new brands from this PR in new branch